src/Featurizers/Components/UnitTests/WindowFeaturizerBase_UnitTest.cpp

```diff
@@ -408,3 +408,48 @@ TEST_CASE("CircularBuffer - range with offset") {
     // when n and offset are not out of bound
     CHECK(++(++start_iter3) == end_iter3);
 }
+
+TEST_CASE("CircularBuffer - clear") {
```

Comment:
I think capacity() should also be tested to make sure capacity is still the same.

Reply:
I am testing it implicitly by adding 6 items and making sure the size still stays at 5. Do you want me to add an explicit test for it as well?

Reply:
I think it would be clearer to have an explicit test to tell we are only resetting the size, not the capacity.

Reply:
Also, capacity is implemented by returning `_maxSize` which is `const` and I can't modify it. I figured a better test instead of checking that would be to make sure the size doesn't go above what it should after I call clear. That is why I didn't have a test that actually calls `circular_buf.capacity()`